File: Java/백준/Silver/9613. GCD 합/GCD 합.java
import java.io.*;

public class Main {
    static InputStreamReader reader = new InputStreamReader(System.in);
    static BufferedReader br = new BufferedReader(reader);

    static OutputStreamWriter writer = new OutputStreamWriter(System.out);
    static BufferedWriter bw = new BufferedWriter(writer);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(br.readLine());
        for (int t = 0; t < T; t++) {
            long gcdSum = 0;
            String inp[] = br.readLine().split(" ");
            int eleCnt = Integer.parseInt(inp[0]);
            int elements[] = new int[eleCnt];
            for (int i = 0; i < eleCnt; i++) elements[i] = Integer.parseInt(inp[1 + i]);
            for (int i = 0; i < eleCnt - 1; i++) {
                for (int j = i + 1; j < eleCnt; j++) {
                    gcdSum += GCD(elements[i], elements[j]);
                }
            }
            bw.write(Long.toString(gcdSum) + "\n");
        }

        bw.flush();
        bw.close();
    }

    static int GCD(int a, int b) {
        if (b == 0) return a;
        else return GCD(b, a % b);
    }
}

File: Python/백준/Silver/9613. GCD 합/GCD 합.py
import sys
T = int(sys.stdin.readline())
for _ in range(T):
    gcd_sum = 0
    inp = tuple(map(int, sys.stdin.readline().split()))
    for i in range(1, len(inp) - 1):
        for j in range(i + 1, len(inp)):
            a, b = inp[i], inp[j]
            while b:
                a, b = b, a % b
            gcd_sum += a
    print(gcd_sum)